actions/watch.ts

```diff
@@ -36,34 +43,51 @@ const GPV2SETTLEMENT = "0x9008D19f58AAbD9eD0D60971565AA8510560ab41";
 export const checkForSettlement: ActionFn = async (
   context: Context,
   event: Event
+) => {
+  return _checkForSettlement(context, event).catch(handleExecutionError);
+};
+
+/**
+ * Asyncronous version of checkForSettlement. It will process all the settlements, and will throw an error at the end if there was at least one error
+ */
+const _checkForSettlement: ActionFn = async (
+  context: Context,
+  event: Event
 ) => {
   const transactionEvent = event as TransactionEvent;
   const settlement = GPv2Settlement__factory.createInterface();
 
-  const registry = await Registry.load(context, transactionEvent.network);
+  const { registry } = await init(transactionEvent.network, context);
```

Comment:
now building the registry is not responsibility of the actions itself

They just need to init the executionContext